core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java

```diff
@@ -128,11 +128,17 @@ public static Environment prepareEnvironment(Settings input, Terminal terminal,
      * Initializes the builder with the given input settings, and loads system properties settings if allowed.
      * If loadDefaults is true, system property default settings are loaded.
      */
-    private static void initializeSettings(Settings.Builder output, Settings input, Map<String, String> esSettings) {
+    static void initializeSettings(Settings.Builder output, Settings input, Map<String, String> esSettings) {
         output.put(input);
         output.putProperties(esSettings,
-            PROPERTY_DEFAULTS_PREDICATE.and(key -> output.get(STRIP_PROPERTY_DEFAULTS_PREFIX.apply(key)) == null),
+                PROPERTY_DEFAULTS_PREDICATE
+                        .and(key -> output.get(STRIP_PROPERTY_DEFAULTS_PREFIX.apply(key)) == null)
+                        .and(key -> output.get(STRIP_PROPERTY_DEFAULTS_PREFIX.apply(key) + ".0") == null),
             STRIP_PROPERTY_DEFAULTS_PREFIX);
+        final String key = Environment.DEFAULT_PATH_DATA_SETTING.getKey();
+        if (esSettings.containsKey(key)) {
```

Comment:
Probably worth a comment for posterity. We want to keep this one to track down an issue caused by 5.3.0. That way come 7 we can remove it.